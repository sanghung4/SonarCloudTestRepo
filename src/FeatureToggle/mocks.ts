import { Feature } from 'generated/graphql';

export const mockFeatures: Feature[] = [
  { id: '1', isEnabled: false, name: 'noooo' },
  { id: '2', isEnabled: true, name: 'fine' },
  { id: '3', isEnabled: true, name: 'test' }
];
