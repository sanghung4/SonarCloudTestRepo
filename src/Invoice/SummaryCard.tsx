import React from 'react';

import {
  Box,
  Card,
  Grid,
  Divider,
  Paper,
  Skeleton,
  Typography,
  alpha
} from '@dialexa/reece-component-library';
import { useTranslation } from 'react-i18next';

import { format } from 'utils/currency';
import { Invoice } from 'generated/graphql';

type Props = {
  loading?: boolean;
  invoice?: Invoice;
};

function SummaryCard(props: Props) {
  /**
   * Custom hooks
   */
  const { t } = useTranslation();

  return (
    <Card sx={{ px: 2, py: 3 }}>
      <Box pb={3} px={3}>
        <Typography
          color="primary"
          variant="h5"
          data-testid="invoice-summary-header"
        >
          {t('invoice.invoiceSummary')}
        </Typography>
      </Box>
      <Paper
        elevation={0}
        sx={{
          px: 2,
          py: 3,
          bgcolor: (theme) => alpha(theme.palette.primary02.main, 0.05)
        }}
      >
        {/* INVOICE SUBTOTAL*/}

        <Grid container px={1}>
          <Grid item xs={6}>
            <Typography color="primary" fontWeight={500}>
              {t('invoice.subtotal')}
            </Typography>
          </Grid>
          <Grid item xs={6}>
            <Typography
              color="primary"
              align="right"
              data-testid="invoice-summary-subtotal"
            >
              {props.loading ? (
                <Box display="flex" justifyContent="flex-end" component="span">
                  <Skeleton width={80} />
                </Box>
              ) : (
                format(parseFloat(props?.invoice?.subtotal ?? '0'))
              )}
            </Typography>
          </Grid>
          <Grid item xs={6}>
            <Typography color="primary">{t('common.tax')}</Typography>
          </Grid>
          <Grid item xs={6}>
            <Typography
              color="primary"
              align="right"
              data-testid="invoice-summary-tax"
            >
              {props.loading ? (
                <Box display="flex" justifyContent="flex-end" component="span">
                  <Skeleton width={80} />
                </Box>
              ) : (
                format(parseFloat(props?.invoice?.tax ?? '0'))
              )}
            </Typography>
          </Grid>
          <Grid item xs={6}>
            <Typography color="primary">{t('invoice.other')}</Typography>
          </Grid>
          <Grid item xs={6}>
            <Typography
              color="primary"
              align="right"
              data-testid="invoice-summary-other"
            >
              {props.loading ? (
                <Box display="flex" justifyContent="flex-end" component="span">
                  <Skeleton width={80} />
                </Box>
              ) : (
                format(parseFloat(props?.invoice?.otherCharges ?? '0'))
              )}
            </Typography>
          </Grid>
        </Grid>
        <Box my={2}>
          <Divider />
        </Box>

        {/*TOTAL AMOUNT AND PAID TO DATE*/}

        <Grid container spacing={2} px={1}>
          <Grid item xs={6}>
            <Typography color="primary" fontWeight={500}>
              {t('invoice.totalAmountDue')}
            </Typography>
          </Grid>
          <Grid item xs={6}>
            <Typography
              color="primary"
              align="right"
              data-testid="invoice-summary-totalAmountDue"
            >
              {props.loading ? (
                <Box display="flex" justifyContent="flex-end" component="span">
                  <Skeleton width={80} />
                </Box>
              ) : (
                format(parseFloat(props?.invoice?.originalAmt ?? '0'))
              )}
            </Typography>
          </Grid>
          <Grid item xs={6}>
            <Typography color="primary" sx={{ fontWeight: 500 }}>
              {t('invoice.paidToDate')}
            </Typography>
          </Grid>
          <Grid item xs={6}>
            <Typography
              color="primary"
              align="right"
              data-testid="invoice-summary-paidToDate"
            >
              {props.loading ? (
                <Box display="flex" justifyContent="flex-end" component="span">
                  <Skeleton width={80} />
                </Box>
              ) : (
                format(parseFloat(props?.invoice?.paidToDate ?? '0'))
              )}
            </Typography>
          </Grid>
        </Grid>
        <Box my={2}>
          <Divider />
        </Box>

        {/**REMAINING AMOUNT */}

        <Grid container spacing={2} px={1}>
          <Grid item xs={6}>
            <Typography color="primary" fontWeight={500}>
              {t('invoice.remainingAmount')}
            </Typography>
          </Grid>
          <Grid item xs={6}>
            <Typography
              color="primary"
              align="right"
              display="flex"
              justifyContent="flex-end"
              data-testid="invoice-summary-remainingAmount"
            >
              {props.loading ? (
                <Skeleton width={80} />
              ) : (
                format(parseFloat(props?.invoice?.openBalance ?? '0'))
              )}
            </Typography>
          </Grid>
        </Grid>
      </Paper>
    </Card>
  );
}

export default SummaryCard;
