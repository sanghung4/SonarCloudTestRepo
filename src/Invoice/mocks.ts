import { GetInvoiceDocument } from 'generated/graphql';

export const successMincron = {
  request: {
    query: GetInvoiceDocument,
    variables: {
      accountId: '123456',
      invoiceNumber: '5460026'
    }
  },
  result: {
    data: {
      invoice: {
        invoiceNumber: '5460026',
        status: 'Open',
        terms: 'NET 30 DAYS',
        customerPo: '40251/13325',
        invoiceDate: '10/28/2020',
        dueDate: '10/28/2020',
        originalAmt: '',
        openBalance: '',
        age: '',
        jobNumber: 'AVLNEP7',
        jobName: 'NEP 7 ASHEVILLE',
        address: {
          streetLineOne: '911 BLUEBERRY WAY',
          streetLineTwo: '',
          streetLineThree: '',
          city: 'NORTHLAKE',
          state: 'TX',
          postalCode: '76226',
          country: ''
        },
        shipDate: '11/06/2020',
        deliveryMethod: 'Our Truck',
        subtotal: '3184.28',
        tax: '262.76',
        otherCharges: '0.0',
        paidToDate: '1000.00',
        invoiceItems: [
          {
            id: '1738433F4-91A7-4637-8C60-B4DF59A060C1',
            brand: 'Advanced Drainage Systems, Inc.',
            name: '120 x 7.5, 100 Sq Yd, Geosynthetic, Double, Photodegradable, Erosion Control Blanket',
            partNumber: '8X1125STRAWDN',
            mfr: '00S2TT',
            thumb: 'null',
            price: '43.59',
            qty: {
              quantityOrdered: 1,
              quantityShipped: 1
            }
          }
        ]
      }
    }
  }
};

export const successMincron_5448540 = {
  request: {
    query: GetInvoiceDocument,
    variables: {
      accountId: '91492',
      invoiceNumber: '5448540'
    }
  },
  result: {
    data: {
      invoice: {
        invoiceNumber: '5282411',
        status: 'Open',
        terms: 'NET 30 DAYS',
        customerPo: 'WILKESBORO JIM',
        invoiceDate: '10/06/2021',
        dueDate: '10/28/2020',
        originalAmt: '',
        openBalance: '',
        age: '',
        jobNumber: 'SHOP',
        jobName: 'SHOP',
        address: {
          streetLineOne: '911 BLUEBERRY WAY',
          streetLineTwo: '',
          streetLineThree: '',
          city: 'NORTHLAKE',
          state: 'TX',
          postalCode: '76226',
          country: ''
        },
        shipDate: '11/06/2021',
        deliveryMethod: 'Our Truck',
        subtotal: '2184.28',
        tax: '162.73',
        otherCharges: '0.0',
        paidToDate: '139.91',
        invoiceItems: {
          id: 'F02FD85F-4DCE-4EA7-8459-2DA7CFD689DC',
          brand: 'Certex USA, name:2 x 12, Nylon, Eye and Eye, Ply Sling',
          partNumber: 'SLING212,mfr:CX07-0126-12',
          thumb:
            'https://images.tradeservice.com/ProductImages/DIR100193/CERTEX_CX07-0126-12_LINE_SML.jpg',
          price: '40.00',
          qty: {
            quantityOrdered: 1,
            quantityShipped: 1
          }
        }
      }
    }
  }
};
