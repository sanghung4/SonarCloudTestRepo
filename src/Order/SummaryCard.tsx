import { useCallback, useMemo } from 'react';

import {
  Box,
  Button,
  Card,
  CircularProgress,
  Grid,
  Divider,
  Paper,
  Skeleton,
  Typography,
  alpha,
  useSnackbar,
  useTheme
} from '@dialexa/reece-component-library';
import { useTranslation } from 'react-i18next';

import { Order, ProductPricing } from 'generated/graphql';
import addItemsToCartCb from 'Order/utils/addItemsToCartCb';
import { format } from 'utils/currency';
import { useCartContext } from 'providers/CartProvider';

type Props = {
  loading?: boolean;
  order?: Order;
  pricingData: ProductPricing[];
};

function SummaryCard(props: Props) {
  /**
   * Custom hooks
   */
  const { t } = useTranslation();
  const theme = useTheme();
  const { pushAlert } = useSnackbar();

  /**
   * Context
   */
  const { addItemsToCart, cartLoading } = useCartContext();

  /**
   * Memo
   */
  const hasItemsAvailableForReorder = useMemo(hasItemsAvailableForReorderMemo, [
    props
  ]);

  /**
   * Callbacks
   */
  const handleAddItemsToCart = useCallback(handleAddItemsToCartCb, [
    addItemsToCart,
    props.order,
    pushAlert,
    props.pricingData
  ]);

  return (
    <Card sx={{ px: 2, py: 3 }}>
      <Box pb={3} px={3}>
        <Typography color="primary" variant="h5">
          {t('cart.orderSummary')}
        </Typography>
      </Box>
      <Paper
        elevation={0}
        sx={{
          px: 2,
          py: 3,
          bgcolor: (theme) => alpha(theme.palette.primary02.main, 0.05)
        }}
      >
        <Grid container px={1}>
          <Grid item xs={6}>
            <Typography color="primary" fontWeight={500}>
              {t('cart.subtotal')}
            </Typography>
          </Grid>
          <Grid item xs={6}>
            <Typography
              color="primary"
              align="right"
              data-testid="order-summary-subtotal"
            >
              {props.loading ? (
                <Box display="flex" justifyContent="flex-end" component="span">
                  <Skeleton width={80} />
                </Box>
              ) : (
                format(props.order?.subTotal ?? 0)
              )}
            </Typography>
          </Grid>
          <Grid item xs={6}>
            <Typography color="primary">{t('common.tax')}</Typography>
          </Grid>
          <Grid item xs={6}>
            <Typography
              color="primary"
              align="right"
              data-testid="order-summary-tax"
            >
              {props.loading ? (
                <Box display="flex" justifyContent="flex-end" component="span">
                  <Skeleton width={80} />
                </Box>
              ) : (
                format(props.order?.tax ?? 0)
              )}
            </Typography>
          </Grid>
        </Grid>
        <Box my={2}>
          <Divider />
        </Box>
        <Grid container spacing={2} px={1}>
          <Grid item xs={6}>
            <Typography color="primary" fontWeight={500}>
              {t('orders.orderTotal')}
            </Typography>
          </Grid>
          <Grid item xs={6}>
            <Typography
              color="primary"
              align="right"
              display="flex"
              justifyContent="flex-end"
              data-testid="order-summary-order-total"
            >
              {props.loading ? (
                <Skeleton width={80} />
              ) : (
                format(
                  props.order?.orderTotal ??
                    (props.order?.subTotal ?? 0) + (props.order?.tax ?? 0)
                )
              )}
            </Typography>
          </Grid>
          <Grid item xs={12}>
            <Button
              disabled={
                cartLoading || props.loading || !hasItemsAvailableForReorder
              }
              variant="primary"
              size="small"
              fullWidth
              onClick={handleAddItemsToCart}
              data-testid="add-all-to-cart-button"
            >
              {(cartLoading || props.loading) && (
                <Box display="inline-block" mr={1}>
                  <CircularProgress size={theme.spacing(2)} />
                </Box>
              )}
              {t('orders.addAll')}
            </Button>
          </Grid>
        </Grid>
      </Paper>
    </Card>
  );

  function handleAddItemsToCartCb() {
    addItemsToCartCb({
      order: props.order,
      pushAlert,
      addItemsToCart,
      pricingData: props.pricingData
    });
  }

  function hasItemsAvailableForReorderMemo() {
    if (props.order?.lineItems?.length) {
      const items = props.order.lineItems.filter((i) => {
        const pricing = props.pricingData.filter(
          (price) => price.productId === i.erpPartNumber
        );
        return !!pricing?.[0]?.sellPrice;
      });
      return items.length > 0;
    }
  }
}

export default SummaryCard;
