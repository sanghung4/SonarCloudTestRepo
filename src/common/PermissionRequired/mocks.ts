import { UserContextType } from 'AuthProvider';
import { Permission } from 'common/PermissionRequired';

export const authConfigNoPerm = {
  authState: {
    isAuthenticated: true
  },
  profile: {
    isEmployee: true,
    isVerified: true
  } as UserContextType['profile'],
  activeFeatures: ['LISTS']
};
export const authConfig = {
  authState: {
    isAuthenticated: true
  },
  profile: {
    userId: 'test',
    permissions: [Permission.EDIT_LIST, Permission.APPROVE_CART],
    isEmployee: true,
    isVerified: true
  } as UserContextType['profile'],
  activeFeatures: ['LISTS']
};
