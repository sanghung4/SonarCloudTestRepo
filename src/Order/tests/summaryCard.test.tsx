import { fireEvent } from '@testing-library/react';

import { mockCartContext } from 'Cart/tests/mocks';
import SummaryCard from 'Order/SummaryCard';
import { orderMocks, pricingDataMocks2 } from 'Order/tests/mocks';
import { CartContext } from 'providers/CartProvider';
import { render } from 'test-utils/TestWrapper';

/**
 * Mock values
 */
const mockProps: Parameters<typeof SummaryCard>[0] = {};
const mocks = {
  cart: { ...mockCartContext },
  props: { ...mockProps }
};

/**
 * Mock methods
 */
jest.mock('@dialexa/reece-component-library', () => ({
  ...jest.requireActual('@dialexa/reece-component-library'),
  __esModule: true,
  useSnackbar: () => ({ pushAlert: jest.fn() })
}));

/**
 * Setup test function
 */
function setup(m: typeof mocks) {
  return render(<SummaryCard {...m.props} />, { cartConfig: m.cart });
}

/**
 * TEST
 */
describe('Order - SummaryCard', () => {
  afterEach(() => {
    mocks.cart = { ...mockCartContext };
    mocks.props = {};
  });

  it('Should match snapshot when loading', () => {
    mocks.props.loading = true;
    const { container } = setup(mocks);
    expect(container).toMatchSnapshot();
  });

  it('Should match snapshot with undefined order data', () => {
    const { container } = setup(mocks);
    expect(container).toMatchSnapshot();
  });

  it('Should match snapshot with an order', () => {
    mocks.props.order = { ...orderMocks[0] };
    mocks.props.pricingData = pricingDataMocks2;
    const { container } = setup(mocks);
    expect(container).toMatchSnapshot();
  });

  it('Should expect addToCart is not called with empty pricing data', () => {
    mocks.props.order = { ...orderMocks[0] };
    mocks.props.pricingData = [];
    const { getByTestId } = setup(mocks);
    fireEvent.click(getByTestId('add-all-to-cart-button'));
    expect(mocks.cart.addItemsToCart).toBeCalledTimes(0);
  });

  it('Should expect addToCart is called', () => {
    mocks.props.order = { ...orderMocks[0] };
    mocks.props.pricingData = pricingDataMocks2;
    const { getByTestId } = setup(mocks);
    fireEvent.click(getByTestId('add-all-to-cart-button'));
    expect(mocks.cart.addItemsToCart).toBeCalledTimes(0);
  });
});
