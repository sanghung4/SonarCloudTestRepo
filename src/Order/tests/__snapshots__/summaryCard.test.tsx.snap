// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order - SummaryCard Should match snapshot when loading 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiCard-root css-26ymy3-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiBox-root css-ju4zfe"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-1bvxf5q-MuiTypography-root"
      >
        Order Summary
      </h5>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-3zd3l8-MuiPaper-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-cqjonc-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1vcav5x-MuiTypography-root"
          >
            Subtotal
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight css-1ufy15n-MuiTypography-root"
            data-testid="order-summary-subtotal"
          >
            <span
              class="MuiBox-root css-1bvc4cc"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
                style="width: 80px;"
              />
            </span>
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1gzv4l7-MuiTypography-root"
          >
            Tax
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight css-1ufy15n-MuiTypography-root"
            data-testid="order-summary-tax"
          >
            <span
              class="MuiBox-root css-1bvc4cc"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
                style="width: 80px;"
              />
            </span>
          </p>
        </div>
      </div>
      <div
        class="MuiBox-root css-1ebprri"
      >
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-di86mu-MuiDivider-root"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1k5xob2-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1vcav5x-MuiTypography-root"
          >
            Order Total
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight css-1eszk6v-MuiTypography-root"
            data-testid="order-summary-order-total"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
              style="width: 80px;"
            />
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-49904w-MuiGrid-root"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-disableElevation MuiButton-fullWidth MuiButtonBase-root Mui-disabled  css-1fitk8o-MuiButtonBase-root-MuiButton-root"
            data-testid="add-all-to-cart-button"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <div
              class="MuiBox-root css-48y3p0"
            >
              <div
                class="MuiBox-root css-hyxlzm"
              >
                <span
                  aria-valuenow="100"
                  class="MuiCircularProgress-root MuiCircularProgress-determinate MuiCircularProgress-colorPrimary css-1wyqyef-MuiCircularProgress-root"
                  role="progressbar"
                  style="width: 16px; height: 16px; transform: rotate(-90deg);"
                >
                  <svg
                    class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                    viewBox="22 22 44 44"
                  >
                    <circle
                      class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate css-oxts8u-MuiCircularProgress-circle"
                      cx="44"
                      cy="44"
                      fill="none"
                      r="19.5"
                      stroke-width="5"
                      style="stroke-dasharray: 122.522; stroke-dashoffset: 0.000px;"
                    />
                  </svg>
                </span>
                <span
                  class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1yx9kzf-MuiCircularProgress-root"
                  role="progressbar"
                  style="width: 16px; height: 16px;"
                >
                  <svg
                    class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                    viewBox="22 22 44 44"
                  >
                    <circle
                      class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                      cx="44"
                      cy="44"
                      fill="none"
                      r="19.5"
                      stroke-width="5"
                    />
                  </svg>
                </span>
              </div>
            </div>
            Add All to Cart
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Order - SummaryCard Should match snapshot with an order 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiCard-root css-26ymy3-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiBox-root css-ju4zfe"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-1bvxf5q-MuiTypography-root"
      >
        Order Summary
      </h5>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-3zd3l8-MuiPaper-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-cqjonc-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1vcav5x-MuiTypography-root"
          >
            Subtotal
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight css-1ufy15n-MuiTypography-root"
            data-testid="order-summary-subtotal"
          >
            $963.57
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1gzv4l7-MuiTypography-root"
          >
            Tax
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight css-1ufy15n-MuiTypography-root"
            data-testid="order-summary-tax"
          >
            $79.64
          </p>
        </div>
      </div>
      <div
        class="MuiBox-root css-1ebprri"
      >
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-di86mu-MuiDivider-root"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1k5xob2-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1vcav5x-MuiTypography-root"
          >
            Order Total
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight css-1eszk6v-MuiTypography-root"
            data-testid="order-summary-order-total"
          >
            $1,043.21
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-49904w-MuiGrid-root"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-disableElevation MuiButton-fullWidth MuiButtonBase-root Mui-disabled  css-1fitk8o-MuiButtonBase-root-MuiButton-root"
            data-testid="add-all-to-cart-button"
            disabled=""
            tabindex="-1"
            type="button"
          >
            Add All to Cart
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Order - SummaryCard Should match snapshot with undefined order data 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiCard-root css-26ymy3-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiBox-root css-ju4zfe"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-1bvxf5q-MuiTypography-root"
      >
        Order Summary
      </h5>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-3zd3l8-MuiPaper-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-cqjonc-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1vcav5x-MuiTypography-root"
          >
            Subtotal
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight css-1ufy15n-MuiTypography-root"
            data-testid="order-summary-subtotal"
          >
            $0.00
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1gzv4l7-MuiTypography-root"
          >
            Tax
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight css-1ufy15n-MuiTypography-root"
            data-testid="order-summary-tax"
          >
            $0.00
          </p>
        </div>
      </div>
      <div
        class="MuiBox-root css-1ebprri"
      >
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-di86mu-MuiDivider-root"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1k5xob2-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1vcav5x-MuiTypography-root"
          >
            Order Total
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1hhgieh-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight css-1eszk6v-MuiTypography-root"
            data-testid="order-summary-order-total"
          >
            $0.00
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-49904w-MuiGrid-root"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-disableElevation MuiButton-fullWidth MuiButtonBase-root Mui-disabled  css-1fitk8o-MuiButtonBase-root-MuiButton-root"
            data-testid="add-all-to-cart-button"
            disabled=""
            tabindex="-1"
            type="button"
          >
            Add All to Cart
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
